packages/nimble-components/src/table/index.ts

```diff
@@ -177,10 +168,17 @@ export class Table<
     @observable
     public firstSortedColumn?: TableColumn;
 
+    /**
+     * @internal
+     */
+    @observable
+    public documentShiftKeyDown = false;
```

Comment:
It is also used to affect the styling of the table (which is why it is observable) so that all the text doesn't become selected during a SHIFT+CLICK interaction. Since the styling isn't specific to selection, I didn't think it made sense to push the logic into a selection manager.

I was going to point this out to @msmithNI because he was running into a similar problem in #854 where the header text was become selected during a SHIFT+CLICK interaction for sorting. If we're happy with this solution, then I think this code would also solve his problem.